### DISCUSSION TITLE
pump float value

### DISCUSSION THREAD

**Discussion by jianlingdong** [nlohmann/json#2448] [Q&A] [answered]
JSON version 3.9.1

	float f1 = 34.76;
	double d1 = 34.76;
	nlohmann::json  jsonObj;
	jsonObj["f1"] = f1;
	jsonObj["d1"] = d1;
	string str1 = jsonObj.dump();
	cout << str1 << endl; // {"d1":34.76, "f1" : 34.7599983215332}

str1 value is {"d1":34.76, "f1" : 34.7599983215332} ，can i get the value  {"d1":34.76, "f1" : 34.76}。

**Comment by gregmarr** [comment, marked as answer]
A variable of type `float` is unable to store the value `34.76`.  The line `float f1 = 34.76` actually stores the value `34.759998321533203125` in `f1`.  The value has lost precision even before it gets to the `json` library.

**Reply by nlohmann** [reply]
Related:

- https://float.exposed/0x420b0a3d
- https://float.exposed/0x40416147ae147ae1

**Reply by JHerrKodo** [reply]
What is the recommanded way to minimize the output string, or what opportunities are available?
Casting to double does not help. Storing the floating point value as string is possible, but not very nice.
Is it possible to limit the number of decimal places?

**Reply by nlohmann** [reply]
No, there is currently no way to configure this.
